src/query-builder/QueryBuilder.ts

```diff
@@ -932,7 +933,11 @@ export abstract class QueryBuilder<Entity> {
             case "equal":
                 return `${aliasPath} = ${parameters[0]}`;
             case "ilike":
-                return `${aliasPath} ILIKE ${parameters[0]}`;
+                if (this.connection.driver instanceof PostgresDriver) {
+                    return `${aliasPath} ILIKE ${parameters[0]}`;
+                }
+
+                return `UPPER(${aliasPath}) LIKE UPPER(${parameters[0]})`;
```

Comment:
I believe ANSI SQL supports the `COLLATE` operator - which may be more likely to hit an index than this?

For example, SQLite supports something along the lines of..

```
foo COLLATE NOCASE LIKE "test%"
```

Would something like this be more performant?  Is it portable?  (I don't know the answer to these :smile:)

Reply:
Doesn't look good, I don't know if it could be more performant, but it sound like we're lying to the dev that just wanted to do `x ILIKE y`.

Reply:
Also, this is exactly how hibernate works:

https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/criterion/LikeExpression.java

https://github.com/hibernate/hibernate-orm/blob/master/hibernate-core/src/main/java/org/hibernate/criterion/IlikeExpression.java

Reply:
Ah, ok!  Great!  I'm all on board then with `lower` - I'd just remembered seeing someone do that for MSSQL once upon a time and wanted to bring it up :)